test/merge.test.ts

```diff
@@ -90,4 +90,38 @@ describe("endpoint.merge()", () => {
       }
     });
   });
+
+  it("does not mutate the route param", () => {
+    const route = {
+      owner: "octokit",
+      repo: "endpoint.js"
+    };
+
+    endpoint.merge(route);
+
+    expect(route).toEqual({
```

Comment:
Maybe better to use [`.toStrictEqual()`](https://jestjs.io/docs/en/expect#tostrictequalvalue)?

```suggestion
    expect(route).toStrictEqual({
```

Reply:
That sounds good to me!